http/cookies.lua

```diff
@@ -20,15 +26,16 @@ local function parse_set_cookie(text_cookie, host, path, time)
 	}
 	local age = matched_cookie["max-age"]
 	if age then
-		assert(age:match("^-?%d+$"), "expected [-]DIGIT* for max-age field")
-		local negative = age:match("^-")
-		if negative then
+		local is_negative, match = age:match("^(-?)(%d+)$")
```

Comment:
`-` needs to be escaped in a pattern.

Reply:
It's for anchoring, `^` is not a valid character in the RFC.

Reply:
huh? what RFC?